core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -433,21 +433,25 @@ object AdminUtils extends Logging with AdminUtilities {
                   replicationFactor: Int,
                   topicConfig: Properties = new Properties,
                   rackAwareMode: RackAwareMode = RackAwareMode.Enforced) {
+    if (topicExists(zkUtils, topic)) {
+      throw new TopicExistsException(s"Topic '$topic' already exists.")
+    }
     val brokerMetadatas = getBrokerMetadatas(zkUtils, rackAwareMode)
     val replicaAssignment = AdminUtils.assignReplicasToBrokers(brokerMetadatas, partitions, replicationFactor)
-    AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(zkUtils, topic, replicaAssignment, topicConfig)
+    AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(zkUtils, topic, replicaAssignment, topicConfig, checkTopicExistence = false)
```

Comment:
Hmm, I wonder if we should have a `createTopics` method that does `getBrokerMetadatas` once instead of once per topic? This would improve the performance of the create topics request.